DbalConsumerHelperTrait.php

```diff
@@ -134,10 +134,22 @@ protected function removeExpiredMessages(): void
             ->setParameter('redelivered', false, DbalType::BOOLEAN)
         ;
 
-        try {
-            $delete->execute();
-        } catch (RetryableException $e) {
-            // maybe next time we'll get more luck
+        $attempts = 5;
```

Comment:
For whatever it's worth, I've done a very similar thing in a different code base years ago. There, it was already retrying up to 4 times and still getting deadlocks. I think I increased it to 12 or so. I'm not saying you need to increase it to any particular number, but be ready to increase this number.

A further observation: this is a similar tactic to increasing a timeout/memory limit on a long-running process...yes, it will reduce errors, but it doesn't really solve the problem. (the problem is that we're using a database as a queue :upside_down_face: )

Reply:
I'm totally aware this is just duct taping a leaky dam, but we need potential solution for yesterday, without changing our infra.